actions/ulog_ingestion/src/ulog_ingestion/__main__.py

```diff
@@ -0,0 +1,304 @@
+
+    Returns:
+    - None
+    """
+    msg_filter = messages.split(",") if messages else None
+    print(ulog_file_path)
+
+    ulog = ULog(ulog_file_path, msg_filter, True)
+
+    # Setup Environment
+    org_id, input_dir, output_dir, topic_delegate, dataset = setup_env()
+    output_dir_path_mcap, output_dir_temp = utils.setup_output_folder_structure(
+        ulog_file_path, input_dir
+    )
+
+    schema_registry_dict = {}
+    schema_checksum_dict = {}
+
+    for key in ulog.message_formats:
+        json_schema_topic = utils.create_json_schema(ulog.message_formats[key].fields)
+        schema_registry_dict[key] = json_schema_topic
+        schema_checksum_dict[key] = utils.compute_checksum(json_schema_topic)
+
+    dataset_relative_path = pathlib.Path(ulog_file_path).relative_to(input_dir)
+    file_record = dataset.get_file_info(dataset_relative_path)
+
+    topic_association = Association(
+        association_id=file_record.file_id, association_type=AssociationType.File
+    )
+
+    start_time = time.time()
+
+    # This is a temporary fix for Multi Information messages with the same name
+    # https://docs.px4.io/main/en/dev_log/ulog_file_format.html#m-multi-information-message
+    # Similar to Plotjuggler approach
+    # TODO: handle this in a better way
+    message_names_with_multi_id_list = list()
+    for d in sorted(ulog.data_list, key=lambda obj: obj.name):
+        if d.multi_id > 0:
+            message_names_with_multi_id_list.append(d.name)
+
+    message_names_with_multi_id_list = list(set(message_names_with_multi_id_list))
+
+    print(f"message_names_with_multi_id_list: {message_names_with_multi_id_list}")
+
+    # Prepare Arguments for Parallel Processing
+    args_list = [
+        (
+            ulog_object,
+            schema_registry_dict,
+            dataset,
+            message_names_with_multi_id_list,
+            schema_checksum_dict,
+            org_id,
+            topic_association,
+            topic_delegate,
+            output_dir_path_mcap,
+            output_dir_temp,
+        )
+        for ulog_object in sorted(ulog.data_list, key=lambda obj: obj.name)
+    ]
+
+    with ProcessPoolExecutor() as executor:
+        executor.map(process_wrapper, args_list)
+
+    end_time = time.time()
+    elapsed_time = end_time - start_time
+    print(f"The topic create function took {elapsed_time} seconds to run.")
+
+
+parser = argparse.ArgumentParser()
+parser.add_argument(
+    "-i",
+    "--input-dir",
+    dest="input_dir",
+    type=pathlib.Path,
+    required=False,
+    help="Directory containing input files to process",
+    default=os.environ.get(actions.InvocationEnvVar.InputDir.value),
+)
+
+parser.add_argument(
+    "-o",
+    "--output-dir",
+    dest="output_dir",
+    type=pathlib.Path,
+    required=False,
+    help="Directory to which to write any output files to be uploaded",
+    default=os.environ.get(actions.InvocationEnvVar.OutputDir.value),
+)
+
+parser.add_argument(
```

Comment:
Yep, this parameter allows the user to specify which topics they want to extract. Thought it might be handy if we want to add support for ingestion configuration (e.g. only ingest image topics etc...).  I renamed "messages" to "topic-names" to make it more clear.